app/services/location_and_outlines.py

```diff
@@ -0,0 +1,148 @@
+"""Module containing service functions for the location and outlines path"""
+
+from app.checks import autocomplete_check, text_search_similarity_check
+from app.config import Config
+from app.google_api_calls import geocoding_api_id, text_search
+from app.helper import (
+    extract_building_polygons,
+    insert_google_data,
+    preprocess_building_outlines,
+)
+
+logger = Config.logger
+
+
+def process_location_and_outlines(session, giving_partner):
+    """Module that processes location and outlines each GP"""
+    logger.info(
+        "Processing location and outline for giving partner",
+        value={
+            "giving_partner_id": str(giving_partner.id),
+        },
+    )
+
+    # Autocomplete checks if Google has matching location details.
+    # If so, we skip the more expensive text search call.
+    if Config.ENABLE_AUTOCOMPLETE:
+        place_id = autocomplete_check(giving_partner)
+        if place_id and process_autocomplete_results(session, giving_partner, place_id):
+            logger.info(
+                "Autocomplete process successful for GP",
+                value={
+                    "giving_partner_id": str(giving_partner.id),
+                    "status": "success",
+                },
+            )
+            return
+
+    text_search_results = text_search(giving_partner)
+    if not len(text_search_results) > 0:
+        logger.info(
+            "No text search results for GP",
+            value={
+                "giving_partner_id": str(giving_partner.id),
+            },
+        )
+        return
+
+    top_text_search_result = text_search_results[0]
+    if text_search_similarity_check(giving_partner, top_text_search_result):
+        process_text_search_results(session, giving_partner, top_text_search_result)
+        logger.info(
+            "Text search process successful for GP",
+            value={"giving_partner_id": str(giving_partner.id), "status": "success"},
+        )
+    else:
+        logger.info(
+            "Top most text search result is not viable for GP",
+            value={
+                "giving_partner_id": str(giving_partner.id),
+                "top_text_search_result": top_text_search_result,
+            },
+        )
+
+    return
+
+
+def process_autocomplete_results(session, giving_partner, place_id):
+    """Handles GP location retrieval using Autocomplete API logic"""
+    gp_address = f"{giving_partner.address}, {giving_partner.city}, {giving_partner.state}, {giving_partner.country}"  # pylint: disable=line-too-long
```

Comment:
If giving_partner.address (or any field) is None, you’ll end up with strings like "None, None, ..." in the address.

You might want to guard or filter:

```suggestion
    gp_address = ", ".join(filter(None, [
    giving_partner.address,
    giving_partner.city,
    giving_partner.state,
    giving_partner.country,
]))
```

Reply:
Good catch, done